applications/FluidDynamicsApplication/custom_utilities/qsvms_dem_coupled_data.h

```diff
@@ -44,6 +44,7 @@ class QSVMSDEMCoupledData : public QSVMSData<TDim, TNumNodes, TElementIntegrates
 
 using NodalScalarData = typename FluidElementData<TDim,TNumNodes, false>::NodalScalarData;
 using NodalVectorData = typename FluidElementData<TDim,TNumNodes, false>::NodalVectorData;
+using NodalTensorData = std::vector< Matrix >;
```

Comment:
If I'm not wrong the sizes of this are known at compile time so stack arrays must be used.

Reply:
Do you mean static arrays?

Reply:
Yes, I think in here you can use `std::array<BoundedMatrix<double,TDim,TDim>,TNumNodes>`. @jgonzalezusua please correct me if I'm wrong.

Reply:
Let me try to change the variable.

Reply:
But I do not see the point in avoiding to write a dynamic template since it does not affect the performance neither the compiler nor computation of the problem.

Reply:
It does affect the performance. `Matrix` is a heap array while `BoundedMatrix` is a stack array, whose allocation is much more efficient. This is why all the arrays in the element are fixed size arrays.

Reply:
Indeed, bounded arrays MUST be used whenever it's possible.

Reply:
Ok